packages/web-components/src/components/cta-section/cta-section.ts

```diff
@@ -62,6 +64,37 @@ class DDSCTASection extends StableSelectorMixin(DDSContentItem) {
       .assignedNodes()
       .some(node => node.nodeType !== Node.TEXT_NODE || node!.textContent!.trim());
     this[slotExistencePropertyNames[name] || '_hasFooter'] = hasContent;
+
+    const childItems = (target as HTMLSlotElement).assignedNodes();
+    const newArray = new Array(childItems.length);
+
+    childItems.forEach((elem, index) => {
+      /* eslint-disable-next-line prefer-destructuring */
+      newArray[index] = elem.childNodes[3].childNodes[2];
+    });
+
+    // @ts-ignore
+    const resizeObserver = new ResizeObserver(entries => {
+      entries.forEach(entry => {
+        entry.target.style.height = 'auto';
+      });
+
+      if (window.innerWidth >= parseFloat(breakpoints.md.width) * baseFontSize) {
+        entries.forEach(entry => {
+          entry.target.style.height = `${Math.max(
+            ...newArray.map(o => {
+              return o.clientHeight;
+            })
+          )}px`;
+        });
+      }
+    });
```

Comment:
The `ResizeObserver` has to be lifecycle-managed (should be gone when the element goes out of the render tree). Please see https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/176b82ad97cc9cbb7302ce4f8ebc281eef6fca93/packages/web-components/src/components/carousel/carousel.ts#L200-L212 for more details.

Reply:
Thanks for the example! I think I managed to implement the lifecycle management correctly, let me know how it looks!